### PR TITLE
Implemented SESSION_COOKIE_SAMESITE

### PR DESCRIPTION
**Pull request by tylersalminen**

### CODE REVIEW

File: flask_session/sessions.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def save_session(self, app, session, response):
 
         httponly = self.get_cookie_httponly(app)
         secure = self.get_cookie_secure(app)
+        samesite = self.get_cookie_samesite(app)
```
Comment (marius56782) [inline flask_session/sessions.py]: Where is the definition of ``get_cookie_samesite()``?

Comment (tylersalminen) [reply flask_session/sessions.py]: Its implemented in flask.sessions.SessionInterface of main Flask repository

Comment (marius56782) [reply flask_session/sessions.py]: I see. But it doesn't look like anyone is maintaining the flask-session repo? Last commit was in 2017 as far as I can see

Comment (tylersalminen) [reply flask_session/sessions.py]: Yea that's what I was worried about, I may just use my own fork. The feature is not critical for the project I'm working on, just nice to have.

Comment (marius56782) [reply flask_session/sessions.py]: Yes, would be really nice to have it

**Comment by iTaybb** [comment]
Would like to see this merged

**Comment by brunlid** [comment]
would be great to see this merged please
```

        httponly = self.get_cookie_httponly(app)
        secure = self.get_cookie_secure(app)
        expires = self.get_expiration_time(app, session)
        val = self.serializer.dumps(dict(session))
        samesite = self.get_cookie_samesite(app)
        self.redis.setex(name=self.key_prefix + session.sid, value=val,
                         time=total_seconds(app.permanent_session_lifetime))
        if self.use_signer:
            session_id = self._get_signer(app).sign(want_bytes(session.sid))
        else:
            session_id = session.sid
        response.set_cookie(app.session_cookie_name, session_id,
                            expires=expires, httponly=httponly,
                            domain=domain, path=path, secure=secure,
                            samesite=samesite)
```

**Comment by fengsp** [comment]
Versions of Flask older than `1.0` would break.

**Comment by eelkevdbos** [comment]
@fengsp Would you be willing to merge if we created some kind of capabilities switch based on flask version?

**Comment by icappello** [comment]
Using some try-except blocks could be a rather quick solution to avoid breaking older Flask versions: try calling the `set_cookie` method with the `samesite` parameter and fallback to the current call in case of exceptions

**Comment by tylersalminen** [comment]
Ok sorry have not been looking at this, I updated my fork and commited and update that will conditionally resolve and pass the samesite cookie kwarg to the set_cookie method on the SessionInterface base. Should be compatible with older versions of flask that do not have get_cookie_samsite and/or do not have the samesite kwarg on set_cookie

**Comment by ChangYuJonathanWu** [comment]
I think it's pretty critical to get this fix in. Without it, the Flask `SESSION_COOKIE_SAMESITE` setting essentially goes into a void and does so silently. As others have mentioned, browsers have implemented default controls around the SameSite attribute. Unless developers are checking their session cookie manually, they may not even notice their security settings aren't being propagated. This is on-top of potentially breaking applications due to browser enforcement of cookie attributes.

This fix looks good, and I'm hoping it can be merged in soon.

**Comment by scarpenter-avenues** [comment]
This would be super helpful.  This issue is breaking an app I am developing for my school.

**Comment by benshanahan1** [comment]
+1 This should be merged, especially with Chrome's recent removal of 3rd party cookie support with `SameSite=None`.

**Pull request merged**